tests/unit_tests/commandline/test_geoips_list_scripts.py

```diff
@@ -0,0 +1,115 @@
+"""
+
+from glob import glob
+from importlib import resources
+from os.path import basename
+import pytest
+
+from tests.unit_tests.commandline.cli_top_level_tester import BaseCliTest
+
+
+class TestGeoipsListScripts(BaseCliTest):
+    """Unit Testing Class for GeoipsListScripts Command."""
+
+    @property
+    def all_possible_subcommand_combinations(self):
+        """A list of every possible call signature for the GeoipsListScripts command.
+
+        This includes failing cases as well.
+        """
+        if not hasattr(self, "_cmd_list"):
+            self._cmd_list = []
+            base_args = self._list_scripts_args
+            for pkg_name in self.plugin_packages + ["all"]:
+                if pkg_name != "all":
+                    args = base_args + ["-p", pkg_name]
+                else:
+                    args = base_args
+                self._cmd_list.append(args)
+            # Add argument list to retrieve help message
+            self._cmd_list.append(base_args + ["-h"])
+            # Add argument list with a non-existent package
+            self._cmd_list.append(base_args + ["-p", "non_existent_package"])
+        return self._cmd_list
+
+    def check_error(self, args, error):
+        """Ensure that the 'geoips list scripts ...' error output is correct.
+
+        Parameters
+        ----------
+        args: 2D list of str
+            - The arguments used to call the CLI (expected to fail)
+        error: str
+            - Multiline str representing the error output of the CLI call
+        """
+        # An error occurred using args. Assert that args is not valid and check the
+        # output of the error.
+        assert args != ["geoips", "list", "scripts"]
+        for pkg_name in self.plugin_packages:
+            assert args != ["geoips", "list", "scripts", "-p", pkg_name]
+        assert "usage: To use, type `geoips list scripts`" in error
+
+    def check_output(self, args, output):
+        """Ensure that the 'geoips list scripts ...' successful output is correct.
+
+        Parameters
+        ----------
+        args: 2D list of str
+            - The arguments used to call the CLI
+        output: str
+            - Multiline str representing the output of the CLI call
+        """
+        # The args provided are valid, so test that the output is actually correct
+        if "-h" in args:
+            assert "usage: To use, type `geoips list scripts`" in output
+        else:
+            # Checking tabular output from the list-scripts command
+            if "-p" in args:
+                # A certain package was requested, generate a list of all scripts from
+                # that package.
+                pkg_names = [args[-1]]
+            else:
+                # no `-p` flag, check all packages instead.
+                pkg_names = self.plugin_packages
+            script_names = []
+            for pkg_name in pkg_names:
+                script_names += sorted(
```

Comment:
It would be my preference to use `.extend` instead of `+=` here. Do we have a convention?

Reply:
@biosafetylvl5 Can I ask why you have this preference? I'm not disagreeing, just wondering. 

It's probably more explicit and tells the reader "this is a list that is being extended" rather than leaving them wondering "what variable type is being added to here?"

Reply:
See #541.